src/lib.rs

```diff
@@ -16,21 +27,67 @@ const ANY_IN_ERROR_MSG: &str = "Resource misses at least one environment variabl
 const ANY_NOT_IN_ERROR_MSG: &str =
     "Resource should not contain any environment variable from the rule.";
 
+const EXTRACT_LABEL_ANNOTATION_REGEX: &str = r"\[.*\]";
+
 #[no_mangle]
 pub extern "C" fn wapc_init() {
     register_function("validate", validate);
     register_function("validate_settings", validate_settings::<settings::Settings>);
     register_function("protocol_version", protocol_version_guest);
 }
 
-fn validate_envvar_with_rule(rule: &Rule, env_vars: &Vec<apicore::EnvVar>) -> Result<()> {
+fn validate_envvar_with_rule(
+    rule: &Rule,
+    pod: &apicore::PodSpec,
+    metadata: &ObjectMeta,
+    env_vars: &Vec<apicore::EnvVar>,
+) -> Result<()> {
     let error_message: &str;
     let mut resource_env_var: HashSet<settings::EnvVar> = HashSet::new();
+    let regex = Regex::new(EXTRACT_LABEL_ANNOTATION_REGEX).unwrap();
     for envvar in env_vars {
-        resource_env_var.insert(settings::EnvVar {
-            name: envvar.name.clone(),
-            value: envvar.value.clone(),
-        });
+        if let Some(env_var_source) = &envvar.value_from {
+            if let Some(field_ref) = &env_var_source.field_ref {
+                let mut value = None;
+                let field_path = field_ref.field_path.as_str();
+                if field_path == "metadata.name" {
+                    value = Some(metadata.name.clone().unwrap());
```

Comment:
I don't like the `unwrap` here. `metadata.name` is already an `Option<String>`, like the `value` variable.
I would just replace the code above with:
```rust
value = metadata.name.clone()
```

Note: I assume `clone` is needed, otherwise just drop it